lib/spack/spack/test/cc.py

```diff
@@ -355,6 +382,26 @@ def test_cc_deps():
             test_args_without_paths)
 
 
+def test_cc_deps_isystem():
```

Comment:
IMO this test feels redundant given `test_cc_deps` and `test_ccld_deps_isystem`. For a given mode I think there could be many possible tests based on the context (e.g. given this test, would it also make sense to add a test when some of the includes are detected to be system directories?).

If this test would have caught an issue you ran into while making this PR, then I think it's worth keeping.